src/vs/workbench/contrib/snippets/browser/snippetsService.ts

```diff
@@ -350,9 +350,16 @@ class SnippetsService implements ISnippetsService {
 	}
 
 	private async _initUserSnippets(): Promise<any> {
-		const userSnippetsFolder = this._userDataProfileService.currentProfile.snippetsHome;
-		await this._fileService.createFolder(userSnippetsFolder);
-		return await this._initFolderSnippets(SnippetSource.User, userSnippetsFolder, this._disposables);
+		const disposables = new DisposableStore();
+		const updateUserSnippets = async () => {
+			disposables.clear();
+			const userSnippetsFolder = this._userDataProfileService.currentProfile.snippetsHome;
+			await this._fileService.createFolder(userSnippetsFolder);
+			await this._initFolderSnippets(SnippetSource.User, userSnippetsFolder, disposables);
+		};
+		this._disposables.add(disposables);
+		this._disposables.add(this._userDataProfileService.onDidChangeCurrentProfile(() => this._pendingWork.push(updateUserSnippets())));
+		updateUserSnippets();
```

Comment:
I believe this first call should awaited because otherwise the `_initUserSnippets` call returns to early

Reply:
It seems no one is awaiting `_initUserSnippets`. But it makes sense to await on first `updateUserSnippets` as `_initUserSnippets` is defined async